src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,72 @@
+package duke.main;
+
+import duke.command.*;
+import duke.exception.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+public class Parser {
+    public static Command parse(String fullCommand) throws DukeException {
+        Command c;
+        String[] splitCommand = fullCommand.split(" ", 2);
+        String headCommand = splitCommand[0];
+        String commandDetails;
+
+        if (headCommand.equals("bye")) return new ExitCommand(null);
+
+        if (headCommand.equals("list")) return new ListCommand(null);
+
+        if (splitCommand.length < 2) throw new EmptyTaskException();
+
+        commandDetails = splitCommand[1];
+        String task;
+        String[] details;
+        int index;
+
+        switch(headCommand) {
+            case "delete":
+                index = Integer.parseInt(commandDetails) - 1;
+                System.out.println(index);
+                return new DeleteCommand(index);
+
+            case "mark":
+                index = Integer.parseInt(commandDetails) - 1;
+                System.out.println(index);
+                return new MarkCommand(index, true);
+
+            case "unmark":
+                index = Integer.parseInt(commandDetails) - 1;
```

Comment:
I like how you used white space around operators. Makes your code look clean and easy to read.